File: 1404. Number of Steps to Reduce a Number in Binary Representation to One.cpp
#include <iostream>
#include <algorithm>
using namespace std;

class Solution
{
public:
    int numSteps(const string& s)
    {
        int n = (int)s.size();
        int ans = n;
        int i = n - 1;
        while (i >= 0 && s[i] == '0')
        {
            i--;
        }
        ans += (i > 0) - (i == 0);
        while (i >= 0)
        {
            ans += s[i] == '0';
            i--;
        }
        return ans;
    }
};

File: 1404. Number of Steps to Reduce a Number in Binary Representation to One.py
class Solution:
    def numSteps(self, s: str) -> int:
        n = len(s)
        ans = 0
        carry = 0
        for i in range(n - 1, 0, -1):
            if s[i] == '0':
                ans += 1 + carry
            else:
                ans += 2 - carry
                carry = 1
        return ans + carry
